### DISCUSSION TITLE
[gatsby-remark-copy-linked-files] A request to make the plugin not restricted to local markdown only

### DISCUSSION THREAD

**Discussion by lee1409** [gatsbyjs/gatsby#31592] [Ideas / Feature Requests] [answered]
It is not a bug but more like a request for making some changes to the code. So... I'm trying to create a [source plugin](https://github.com/lee1409/gatsby-source-yuque) that will fetch markdown (JSON field `string`) remotely from Yuque. Right now, the source plugin able to work with `gatsby-transformer-remark` to create MarkdownRemark node and everything looks fine. However, the `gastby-remark-copy-linked-files` is not able to copy my `svg` files because of this line.

```js
// index.js
if (
      isRelativeUrl(link.url) &&
      getNode(markdownNode.parent).internal.type === `File`
    )
...
```
The reason why it is not working because `gastby-remark-copy-linked-files` will try to check if internal of `MarkdownRemark`'s parent is `File` node, but mine is different which is `YuqueDocumentDetail` node. And I neither want to change my node type nor write every markdown string into local and let gatsby-source-filesystem to read again. That would too much a hassle.

Therefore, I suggest that by editing the line to
```js
if (
      isRelativeUrl(link.url) &&
      getNode(markdownNode.parent).dir
    )
...
```

As long as we have a field called `dir` stating the path of the markdown directory, then the plugin will work for other type of nodes as well. This approach is similar as `gatsby-remark-images` used, therefore I guess there won't be any breaking changes or considerations to change the code.

Hope this gets approved... Thanks! Oh btw, I'm more than happy to contribute if you want me to. 😄

**Comment by lee1409** [comment, marked as answer]
Hello? Anyone?

**Reply by KyleAMathews** [reply]
Hi, yeah the change seems fine to me. A PR would be great!

**Reply by lee1409** [reply]
Alright, thanks! 😃
